modules/travel_pay/app/services/travel_pay/claims_service.rb

```diff
@@ -2,20 +2,38 @@
 
 module TravelPay
   class ClaimsService
-    def get_claims(veis_token, btsss_token)
-      claims_response = client.get_claims(veis_token, btsss_token)
-      symbolized_body = claims_response.body.deep_symbolize_keys
+    def get_claims(veis_token, btsss_token, params = {})
+      faraday_response = client.get_claims(veis_token, btsss_token)
+      raw_claims = faraday_response.body['data'].deep_dup
+
+      claims = filter_by_date(params['appt_datetime'], raw_claims)
 
       {
-        data: symbolized_body[:data].map do |sc|
-          sc[:claimStatus] = sc[:claimStatus].underscore.titleize
+        data: claims.map do |sc|
+          sc['claimStatus'] = sc['claimStatus'].underscore.titleize
           sc
         end
       }
     end
 
     private
 
+    def filter_by_date(date_string, claims)
+      if date_string.present?
+        parsed_appt_date = Date.parse(date_string)
+
+        claims.filter do |claim|
+          !claim['appointmentDateTime'].nil? &&
+            parsed_appt_date == Date.parse(claim['appointmentDateTime'])
```

Comment:
I'm not sure if this is a real concern, but do we want to be more strict in validating `date_string`? I know we're documenting that invalid dates return all claims, but are there inputs where a requester might think they're passing in a date string they think is valid (but it's not) and being confused when all claims are returned?

Reply:
For our immediate use case I think the above is probably fine, but can you please add this to [our tech debt opportunities spreadsheet](https://docs.google.com/spreadsheets/d/1lOHK_cQIEMz85SfFDhyVO42Xq3_g9AFNj9UGYQrPCpI/edit?gid=0#gid=0)?